sources/packages/api/src/main.ts

```diff
@@ -1,20 +1,41 @@
+import "reflect-metadata";
 require("../env_setup");
-import { NestFactory } from "@nestjs/core";
+import { NestFactory, HttpAdapterHost } from "@nestjs/core";
 import { NestExpressApplication } from "@nestjs/platform-express";
 import { ValidationPipe } from "@nestjs/common";
 import { AppModule } from "./app.module";
 import { KeycloakConfig } from "./auth/keycloakConfig";
+import { LoggerService } from "./logger/logger.service";
+import { AppAllExceptionsFilter } from "./app.exception.filter";
+import { exit } from "process";
 
 async function bootstrap() {
   await KeycloakConfig.load();
-  const app = await NestFactory.create<NestExpressApplication>(AppModule);
+  const app = await NestFactory.create<NestExpressApplication>(AppModule, {
+    logger: false,
+  });
+
+  // Setting global prefix
   app.setGlobalPrefix("api");
+
+  // Using LoggerService as app logger
+  app.useLogger(app.get(LoggerService));
+
+  // Exception filter
+  const { httpAdapter } = app.get(HttpAdapterHost);
+  app.useGlobalFilters(new AppAllExceptionsFilter(httpAdapter));
+
+  // Getting application port
   const port = process.env.PORT || 3000;
 
+  // Enabling cors for non prod env
   if (process.env.NODE_ENV !== "production") {
     app.enableCors();
   }
 
+  // Setting express middleware for req
+  app.use(LoggerService.apiLogger);
```

Comment:
Is the idea to log all requests even on PROD env?
Right now we have only conole.log as the final log, but depending on how would be the final log storage maybe we do not want to log every request every time.
Is it a placeholder to be expanded in the future or even be enabled/disabled upon configuration?

Reply:
YES, we need discussion on this topic. And I'm totally agree, logging each request could be overhead if persistent log storage introduce....